feat: add job board section

aaronamano opened the pull request:
# Pull Request
This pull request will tackle and resolve #60 [FEATURE] Job Board

## Description
- I implemented the **Job Board** tab as well as its own section. 
- The user can click on the Job Board tab from the navbar which leads them to the **Job Board** section
- In the section each card is clickable and when the user clicks on the card, the user will be directed a job board url. 
- Designated job urls to its own cards; I obtained the job urls, which were provided in issue #60 
- the design screenshot in issue #60, which shows how the job board should look like, isn’t the same how it looks in this PR as shown under the PR's screenshot

New changes after Greg’s feedback
- Added description under the job board section heading
- Specified aria-labels and made alt tag empty for kite icon
- reduced original sizes of image icons (e.g. 3000 x 2000 px -> 120 x 60 px) and optimized image file storage
- renamed job-board icon files for clarity and convert file type to png

## Type of Change

<!-- Mark the relevant option with an "x" -->

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor

## Related Issues

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [x] Code follows the project's coding standards
- [x] Self-review of the code has been performed
- [x] Code is properly commented where necessary
- [x] No console.log statements left in the code
- [x] No hardcoded values that should be configurable
- [ ] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [x] Changes have been tested locally
- [x] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [x] Manual testing has been performed
- [x] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [ ] All interactive elements are keyboard accessible
- [ ] Color contrast meets WCAG AA standards
- [x] Images have appropriate alt text
- [ ] Form labels are properly associated
- [ ] Focus indicators are visible
- [ ] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [x] Application builds successfully
- [x] No build warnings or errors
- [x] Dependencies are up to date
- [x] No security vulnerabilities introduced

## Screenshots (if applicable)
<img width="1706" height="461" alt="job-board-section-w-description" src="https://github.com/user-attachments/assets/a72fc424-77e2-447b-8827-de6b19c82a12" />
<img width="1104" height="740" alt="job-boards" src="https://github.com/user-attachments/assets/7947a7a1-9e43-40d1-9771-e2e67771e8e3" />

This is what the Job Board section looks like in my PR. You can also see the Job Board tab is highlighted, meaning it is clicked, and the Job Board section appears.


<!-- Add screenshots to help explain your changes -->

## Additional Notes
### Notes for reviewers!
- the design screenshot in issue #60, which shows how the job board should look like, isn’t the same how it looks like in this PR as shown under the PR's screenshot
- let me know if you guys wanna tweak the description of the job board :)
- when clicking on a card in the job board section, it should lead to its valid url/site, please double check and ensure that it does

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.


<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds a Job Board section populated from data, links it from the navbar, and renders it on the Home page.
> 
> - **UI**:
>   - **Job Board**: New `JobBoardSection.jsx` with responsive grid, accessible labels, and sponsor logos/links.
>   - **Navbar**: Enables `jobboard` section link in `src/components/Navbar.jsx`.
>   - **Home**: Renders `JobBoardSection` (2025) in `src/pages/Home.jsx`.
> - **Data**:
>   - Adds `src/data/jobboard.js` with company entries (name, logo, website).
> - **Config**:
>   - Adjusts `jsconfig.json` path mapping to `"@/*": ["./src/*"]`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7c9df0ffba0ea6e23d52d49b12045cdaa3d632a9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

aaronamano committed:
feat: add job board section

shrinkray commented:
@aaronamano could you please write a description for what changes are requested? Please also complete the template. That's helpful to orient the team.

ShugKnight24 commented on jsconfig.json:
```diff
@@ -1,9 +1,8 @@
 {
   "compilerOptions": {
-    "baseUrl": ".",
```
I know baseUrl isn't needed for paths when using TypeScript 4.1, not sure if this applies to jsconfig as well

https://www.typescriptlang.org/tsconfig/#baseUrl

JustinBeaudry replied:
I don't see a reason to remove it, especially in this ticket. I think this is outside the scope of this ticket.

ShugKnight24 commented on src/data/jobboard.js:
```diff
@@ -0,0 +1,52 @@
+    name: 'CCS',
+    logo: ccsLogo,
+    website: 'https://www.ccsdetroit.com/careers/',
+    careersPage: 'https://www.ccsdetroit.com/careers/',
```
This link doesn't exist

the link in the ticket is - https://bit.ly/ccs-jobs-25 which leads to https://www.ccsdetroit.edu/about-us/jobs-at-ccs/

ShugKnight24 commented on src/layouts/JobBoardSection.jsx:
```diff
@@ -0,0 +1,64 @@
+          {year && !isCurrentYear ? `${year} ` : ''}Job Board
+        </h2>
+        <img
+          src={Kite}
```
The image for this section on the mock is the arrow.png, but flipped
I don't have a real preference for this

ShugKnight24 commented on src/layouts/JobBoardSection.jsx:
```diff
@@ -0,0 +1,64 @@
+      </div>
+
+      {/* Job Board Grid - 3x2 layout */}
+      <div className="mx-auto mt-8 max-w-4xl sm:mt-10 md:mt-14 lg:mt-16">
+        <div className="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3">
```
The mockup design looks like it uses the old generic cards

I prefer the look you went with. I would like to see these and the images be a little larger to be more consistent

I'll defer to the group

shrinkray replied:
The site has an issue with core web vitals in that devs have attached huge images then ask the browser to scale the image as needed. I request that we be smarter about attaching images. 

In PR #58 I refactor a component created to be smarter about attaching responsive images. This results in better UX around attached images. For example, the Google logo is huge and by serving a scaled version causes notifications in core web vitals (CWV). Instead of willy nilly attaching whatever images exist, take a look at those. 

In that PR you will see I have replaced images with both better sized and compressed images. In two cases I delivered SVGs in place of images. Use ImageOptim or ImageMagick to help reduce the size of images and if you use the ResponsiveImage component, you may serve both WebP and Png variants which will help hugely. 

We should be adding alt tags or aria-labels on those entities as well.

JustinBeaudry replied:
These are both great points. In the past I've used `imgo` CLI tool for this, but have also built on-the-fly ImageMagick servers to handle this, however I'm out of practice, so I appreciate you two!

aaronamano replied:
Thank you Greg and Justin! I’ll definitely consider this unique practice in the future! :)

ShugKnight24 reviewed (changes requested):
Good work @aaronamano

Request for changes for the CCS careers link

Everything else, I'll defer to @jritten

aaronamano committed:
revert: restore original order of sections and navbar tabs

JustinBeaudry committed:
Merge branch '2025' into 60-add-job-board-feature

JustinBeaudry committed:
Merge branch '2025' into 60-add-job-board-feature

shrinkray committed:
fix: color by context

- navigation for desktop
- nav for mobile

shrinkray committed:
fix: text color

- text color contrast
- button color in contrast
- heading color issues

ShugKnight24 committed:
feat: add initial 2025 speakers

ShugKnight24 committed:
refactor: update `SessionCard` to handle unknown values

- Make aria-controls and aria-label dependant on dynamic data it looks 
for

shrinkray committed:
refactor: allow light/dark modes

allow to inherit colors

ShugKnight24 committed:
refactor: add on file images for prior year speakers

- Optimize assets using imageOptim
- Rename ChrisDemars photo to remove whitespace

ShugKnight24 committed:
chore: correct spelling of `deprecated` directory

ShugKnight24 committed:
refactor: improve styling

- Hardcode colors to be consistent between light and dark mode
- Improve layout of speaker cards by adding margin to the botton and 
centering when mobile

ShugKnight24 committed:
chore: optimize all assets, replace spaces with underscore in name

- optimize all assets with imageOptim
- rename to use underscore, replace name where used

ShugKnight24 committed:
feat: add sponsors23 data to previous events

ShugKnight24 committed:
chore: remove unused image comment

ShugKnight24 committed:
refactor: make all layout headings inline

- Make consistent to other summit website

ShugKnight24 committed:
refactor: make SessionCard data display conditionally

- Improve display so all cards are the same width

shrinkray commented:
@aaronamano   **This request is for this and future PRs.** Go to the top of this and click the '...' button then select Edit. Now you can change this PR. 

Please add a description of what this feature proposed to do. You can get that info from the issue it was built from. When the PR is eventually completed, which issue does it solve.  How will is solve it? What should it look like when done? How should it work? What help do you need that you might not know what to do? Help us, help you.

aaronamano committed:
fix: updated the ccs link and added comment to job board component file

The CCS link initially was invalid and led to a weird site but I resolved it

aaronamano commented:
I fixed my PR description now. I'm sorry about this. I forgot the edit button existed. On a serious note, I would like you guys to view **Notes for reviewers!** under the **Additional Notes** section on top of my PR and help provide your insights please. Thanks :)

shrinkray commented:
@aaronamano Thanks for the update! 

Here's some nits that will improve your branch.  

1. Please merge the current 2025 branch with your branch then update your branch here. If other branches are merged to 2025 you are responsible for keeping your branch updated. 
2. Help save reviewer time.

  - This image is incredibly huge!  See the image below, using the inspection window of the browser, I found this information.  Think about it this way, "What are the sizes I need?" Or after adding it, test to see the size you actually need and replace the one you have. 
  - Please do a better job naming files. Ask, "What is a good name for this file?" 
  - Think about the file type. You can probably get by with a PNG or WebP version of the file. 
  - Use `icn-kite.png` it's the right size 

<img width="549" height="178" alt="lighthouse" src="https://github.com/user-attachments/assets/3974292c-a244-4a1b-9bfb-cde4159d20f2" />
<img width="1022" height="358" alt="inspect" src="https://github.com/user-attachments/assets/33b5fca4-634e-4864-90eb-8d02ed17bc80" />

shrinkray commented:
@aaronamano  I like to see this aria-label, but what is missing is information warning the visitor that this link with take someone off-site. What even is CCS? Please spell out the names. 

- Search for info online for how to write it this ARIA message. 
- Add a section heading and a description for the purpose of this section. This is a great opportunity to add a little marketing spin.  Point out this was made possible by generous sponsors, "...only sponsors at the X level..." get the opportunity to add their links here. You'll have to find out what that is. 
- I see the kite image has an alt tag. It should probably just be empty. It's just decorative, there's no need for someone to learn anything about that graphic. 

<img width="671" height="48" alt="Screenshot 2025-10-10 at 7 31 05 PM" src="https://github.com/user-attachments/assets/3963f22c-1cb4-4211-bcd1-06d644c8553c" />

Keep up the work!

ShugKnight24 committed:
chore: add Michael Bower headshot

aaronamano committed:
fix: renamed image files and made image logo sizes smaller

aaronamano committed:
refactor: tweaked aria labels, replaced svg for png, edited jobboard data

aaronamano committed:
feat: added description in job board section

this is a placeholder but i will change it later

aaronamano committed:
fix: adjusted circle container size and logos

aaronamano committed:
refactor: enhanced job board description

aaronamano commented:
@shrinkray @ShugKnight24 @JustinBeaudry good afternoon! After looking at Greg’s feedback, I added a few changes in my branch. Make sure to relook at the top of my PR as I added a few new stuff.  Please look specifically at the `Description` -> `New changes after Greg’s feedback`, `Notes for reviewers`, and `Screenshots`. Let me know if it needs a few more final changes before it can get merged. Thank you :)

shrinkray commented:
@aaronamano I assigned this to you (above). That's something I do on my cards. It says who is taking responsibility for the branch and let's us all know what's going on. Not everyone does it, but I think it's useful. 

I'll pull down your work and see how you are coming.

ShugKnight24 committed:
Merge branch '2025' into 60-add-job-board-feature

ShugKnight24 commented on src/layouts/JobBoardSection.jsx:
```diff
@@ -0,0 +1,83 @@
+
+      <div className="mx-auto mb-12 max-w-4xl text-center">
+        <p className="mt-6 text-lg leading-8 text-gray-600">
+          This job board is made possible by our generous sponsors spanning from
+          local organizations and regional companies founded in the heart of
+          Detroit to some of the best top global tech companies. They help
+          involuntary provide exclusive opportunities to you, so that you can
+          grow your career in the tech industry. You also get the chance to
+          share your resume online so that our sponsors from Google Developer
+          Groups in Detroit can see your amazing work and be able to reach out
+          to you for exicitng job opportunities to come.
+        </p>
```
Something in this doesn't read right to me

"They help involuntary provide" - I'm not sure, I follow - can you clarify what you mean here

aaronamano replied:
now that i think about it i think the phrase is wordy. by any means i am not an english or marketing major, but i would be happy to change it lol :)

shrinkray replied:
@aaronamano I added comments inline within some of your code but I don't see it showing here. 

- Referencing use of the kite image and styling to improve it Use `icn-kite.png`
- I like having a description, but remove `text-center`. For someone who is dyslectic, it may cause a hardship. Better to give a reference edge. 
- Referencing the use of the `aria-label` attribute, it's somewhat misused
- Referencing the color of the background. you use yellow-200 but I think primary-200 is the same. I suggest using the theme colors when possible.

ShugKnight24 reviewed (commented):
Thanks for adjusting the links

Something about the text in the JobBoard section doesn't make sense to me, though I could be missing it

@shrinkray, offered some additional feedback so I'll defer to him

JustinBeaudry commented on src/layouts/JobBoardSection.jsx:
```diff
@@ -0,0 +1,83 @@
+          grow your career in the tech industry. You also get the chance to
+          share your resume online so that our sponsors from Google Developer
+          Groups in Detroit can see your amazing work and be able to reach out
+          to you for exicitng job opportunities to come.
```
```suggestion
This job board is made possible by our generous sponsors, ranging from local Detroit organizations and regional companies to some of the world's best global tech companies. They voluntarily provide exclusive opportunities so you can grow your tech career. You can share your resume online, allowing sponsors from Google Developer Groups in Detroit to see your work and reach out about exciting job opportunities.
```

JustinBeaudry commented on src/data/jobboard.js:
```diff
@@ -0,0 +1,46 @@
+import grandCircusLogo from '@/assets/images/job-board/grand-circus-logo.png'
+import compassLogo from '@/assets/images/job-board/compass-logo.png'
+import googleLogo from '@/assets/images/job-board/google-logo.png'
+import ibmLogo from '@/assets/images/job-board/ibm-logo.png'
```
My preference is that data be bucketed with the year, as we think about re-usability for next year. @ShugKnight24 @shrinkray thoughts?

ShugKnight24 replied:
No real preference. For consistency and maintainability this makes sense

JustinBeaudry reviewed (changes requested):
I agree with @ShugKnight24 that the wording here doesn't make sense, I've suggested a slight change.

JustinBeaudry committed:
Merge branch '2025' into 60-add-job-board-feature

aaronamano committed:
refactor: fixed a few things from greg and justin feedback

aaronamano commented:
PR should be all set. let me know if i missed anything :)

aaronamano commented:
@JustinBeaudry @shrinkray @ShugKnight24 i tagged you guys just in case you weren't pinged about my previous comment. sorry for the inconvenience.

ShugKnight24 reviewed (commented):
This is looking good @aaronamano, just tested and works well in the browser

Can you take a look @JustinBeaudry's comment about grouping the job board data by year and revert the changes in jsconfig.json

aaronamano committed:
revert: edited jsconfig json file

aaronamano committed:
refactor: moved job board assets to 2025 data

aaronamano committed:
fix: handled syntax error on import statement

aaronamano commented:
ok i updated the branch and made the changes @JustinBeaudry recommended

ShugKnight24 reviewed (approved):
Thanks for the adjustments @aaronamano

LGTM - I'll let @JustinBeaudry take a look

JustinBeaudry reviewed (dismissed):
Code looks great! Nice work @aaronamano! Only issue is that the Grand Circus link 404's. @jritten should we remove it or do you have a difference link for this?
